clang-10 fails to build libdnf: no matching constructor for initialization of 'libdnf::Error'

```
In file included from /usr/ports/sysutils/libdnf/work/libdnf-0.48.0-41-gdf3921b3/libdnf/hy-goal.cpp:40:
In file included from /usr/ports/sysutils/libdnf/work/libdnf-0.48.0-41-gdf3921b3/libdnf/catch-error.hpp:5:
/usr/ports/sysutils/libdnf/work/libdnf-0.48.0-41-gdf3921b3/libdnf/goal/Goal.hpp:39:48: error: no matching constructor for initialization of 'libdnf::Error'
        Error(const char * msg, int errCode) : libdnf::Error(msg), errCode(errCode) {}
                                               ^             ~~~
/usr/ports/sysutils/libdnf/work/libdnf-0.48.0-41-gdf3921b3/libdnf/goal/../error.hpp:14:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'const char *' to 'const libdnf::Error' for 1st argument
class Error : public std::runtime_error {
      ^
/usr/ports/sysutils/libdnf/work/libdnf-0.48.0-41-gdf3921b3/libdnf/goal/../error.hpp:14:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'const char *' to 'libdnf::Error' for 1st argument
/usr/ports/sysutils/libdnf/work/libdnf-0.48.0-41-gdf3921b3/libdnf/goal/../error.hpp:16:5: note: candidate constructor not viable: no known conversion from 'const char *' to 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') for 1st argument
    Error(const std::string & what) : runtime_error(what) {}
    ^
1 error generated.
```

--- a/libdnf/error.hpp
+++ b/libdnf/error.hpp
@@ -13,7 +13,7 @@ namespace libdnf {
  */
 class Error : public std::runtime_error {
 public:
-    Error(const std::string & what) : runtime_error(what) {}
+    using runtime_error::runtime_error;
 };
 
 /**
@@ -33,7 +33,7 @@ public:
  */
 class Exception : public std::runtime_error {
 public:
-    Exception(const std::string & what) : runtime_error(what) {}
+    using runtime_error::runtime_error;
 };
 
 } // namespace libdnf

